__tests__/auth.test.ts

```diff
@@ -1,40 +1,42 @@
-import createJWKSMock from 'mock-jwks';
-import Auth from '../auth';
+import createJWKSMock from 'mock-jwks'
+import { createCheckScopesAndResolve } from '../auth'
 
-import { APIGatewayProxyEvent } from 'aws-lambda';
+import { APIGatewayProxyEvent } from 'aws-lambda'
+
+const TOKEN_ISSUER = 'https://test-app.com/'
 
 describe('Given an event with an [INCORRECT] scope', () => {
   it('Returned - You are not authorized!', async () => {
-    // const jwksMock = await createJwksContext();
-    process.env.JWKS_URI = 'https://test-app.com/.well-known/jwks.json';
-    process.env.TOKEN_ISSUER = 'https://test-app.com/';
-    process.env.AUDIENCE = 'https://test-app.com/test/';
-    const jwksMock = createJWKSMock('https://test-app.com/');
-    await jwksMock.start();
+    const jwksMock = createJWKSMock(TOKEN_ISSUER)
+    await jwksMock.start()
+
+    const checkScopesAndResolve = createCheckScopesAndResolve({
+      jwksUri: 'https://test-app.com/.well-known/jwks.json',
+      issuer: TOKEN_ISSUER,
+      audience: 'https://test-app.com/test/',
+    })
+
     const accessToken = jwksMock.token({
-      aud: [
-        'https://test-app.com/test/'
-      ],
-      iss: 'https://test-app.com/',
+      aud: ['https://test-app.com/test/'],
+      iss: TOKEN_ISSUER,
       sub: 'test-user',
-      scope: 'incorrect scope'
+      scope: 'incorrect scope',
     })
-    console.log('accessToken', accessToken);
-
 
     const mockedEvent = customMockedEvent({
-      authorization: `Bearer ${accessToken}`
-    });
-
-    const authInstance = new Auth();
+      authorization: `Bearer ${accessToken}`,
+    })
 
-    expect(authInstance.checkScopesAndResolve(mockedEvent, ['incorrect scope'])).resolves.toThrow('Error: You are not authorized!');
-    await jwksMock.stop();
+    await expect(
+      checkScopesAndResolve(mockedEvent, ['incorrect scope']),
+    ).rejects.toEqual(new Error('You are not authorized!'))
```

Comment:
You expect this to reject, not to resolve with and `Error`. Resolving with an `Error` is an anti-pattern.